integrations/influxdb_connector/influxdb-timestream-connector/src/timestream_utils.rs

```diff
@@ -88,96 +108,160 @@ pub async fn create_table(
         .send()
         .await?;
 
+    trace!("create_table duration: {:?}", function_start.elapsed());
     Ok(())
 }
 
 pub async fn table_exists(
-    client: &timestream_write::Client,
+    client: &Arc<timestream_write::Client>,
     database_name: &str,
     table_name: &str,
 ) -> Result<bool, Error> {
     // Check if table already exists
 
+    let function_start = Instant::now();
+
     match client
         .describe_table()
         .table_name(table_name)
         .database_name(database_name)
         .send()
         .await
     {
-        Ok(_) => Ok(true),
+        Ok(_) => {
+            trace!("table_exists duration: {:?}", function_start.elapsed());
+            Ok(true)
+        }
         Err(error) => match error
             .as_service_error()
             .map(|e| e.is_resource_not_found_exception())
         {
-            Some(true) => Ok(false),
+            Some(true) => {
+                trace!("table_exists duration: {:?}", function_start.elapsed());
+                Ok(false)
+            }
             _ => Err(anyhow!(error)),
         },
     }
 }
 
 pub async fn database_exists(
-    client: &timestream_write::Client,
+    client: &Arc<timestream_write::Client>,
     database_name: &str,
 ) -> Result<bool, Error> {
     // Check if database already exists
 
+    let function_start = Instant::now();
+
     match client
         .describe_database()
         .database_name(database_name)
         .send()
         .await
     {
-        Ok(_) => Ok(true),
+        Ok(_) => {
+            trace!("database_exists duration: {:?}", function_start.elapsed());
+            Ok(true)
+        }
         Err(error) => match error
             .as_service_error()
             .map(|e| e.is_resource_not_found_exception())
         {
-            Some(true) => Ok(false),
+            Some(true) => {
+                trace!("database_exists duration: {:?}", function_start.elapsed());
+                Ok(false)
+            }
             _ => Err(anyhow!(error)),
         },
     }
 }
 
 pub async fn ingest_records(
-    client: &timestream_write::Client,
-    database_name: &str,
-    table_name: &str,
-    records: &mut [timestream_write::types::Record],
+    client: Arc<timestream_write::Client>,
+    database_name: Arc<String>,
```

Comment:
`ingest_records` is now executed in parallel. Each thread is required to own the arguments it uses.